### PR TITLE
Update docker script

### PR DESCRIPTION
**Pull request by keithc-ca**
* accept JDK versions in any order
* validate requested JDK versions
* get a bootjdk just once (e.g. for jdk17 and jdknext)
* use a Temurin bootjdk 17 until API is available for Semeru

### CODE REVIEW

File: buildenv/docker/mkdocker.sh [hunk 1]
```diff
@@ -29,42 +29,43 @@ usage() {
...
 action=print
+all_versions=
 arch=
-bootjdk_versions=
 cuda=no
 dist=unspecified
 freemarker=no
 gen_git_cache=yes
 jdk_versions=all
+declare -A known_version
```
Comment (dsouzai) [inline buildenv/docker/mkdocker.sh:68]: Probably don't need this to be global considering it's only used in `validate_options`

File: buildenv/docker/mkdocker.sh [hunk 2]
```diff
@@ -582,30 +597,46 @@ install_freemarker() {
...
 
 bootjdk_url() {
   local version=$1
-  if [ $arch = x86_64 ] ; then
-    echo https://api.adoptopenjdk.net/v3/binary/latest/$version/ga/linux/x64/jdk/openj9/normal/adoptopenjdk
+  if [ $version -lt 17 ] ; then
+    echo https://api.adoptopenjdk.net/v3/binary/latest/$version/ga/linux/${arch/x86_64/x64}/jdk/openj9/normal/adoptopenjdk
   else
-    echo https://api.adoptopenjdk.net/v3/binary/latest/$version/ga/linux/$arch/jdk/openj9/normal/adoptopenjdk
+    echo https://api.adoptium.net/v3/binary/latest/$version/ga/linux/${arch/x86_64/x64}/jdk/hotspot/normal/eclipse
```
Comment (dsouzai) [inline buildenv/docker/mkdocker.sh:611]: Kind of a nitpick, but considering `${arch/x86_64/x64}` is used twice, maybe it should just be commoned out into a local var prior to the if/else.

**Review by dsouzai — CHANGES_REQUESTED** [review]
Overall looks good, some minor changes requested.

Also, it maybe worth having a new global variable called 'latest_version' or something along those lines; there's a lot of places where `17` is used directly, which means when the next version comes out, someone's gonna have to go and update all the places.

File: buildenv/docker/mkdocker.sh (hunk 1 above)
Comment (keithc-ca) [reply buildenv/docker/mkdocker.sh:68]: Fixed in f5889f7e1c337a7ff88cd70d2a310e43082790a3.

File: buildenv/docker/mkdocker.sh (hunk 2 above)
Comment (keithc-ca) [reply buildenv/docker/mkdocker.sh:611]: Fixed in f5889f7e1c337a7ff88cd70d2a310e43082790a3.

**Comment by keithc-ca** [comment]
Latest update tweaks comment about source of boot JDKs.

**Comment by dsouzai** [comment]
ran `bash mkdocker.sh --tag=openj9 --dist=ubuntu --version=16.04 --print` before and after

```
[dsouzai@localhost temp]$ diff old new 
126c126
< # Download and install boot JDKs from AdoptOpenJDK.
---
> # Download and install boot JDKs.
130,131c130,131
<  && wget --progress=dot:mega -O jdk16.tar.gz https://api.adoptopenjdk.net/v3/binary/latest/16/ga/linux/x64/jdk/openj9/normal/adoptopenjdk \
<  && mkdir -p /home/jenkins/bootjdks/jdk8 /home/jenkins/bootjdks/jdk11 /home/jenkins/bootjdks/jdk16 \
---
>  && wget --progress=dot:mega -O jdk17.tar.gz https://api.adoptium.net/v3/binary/latest/17/ga/linux/x64/jdk/hotspot/normal/eclipse \
>  && mkdir -p /home/jenkins/bootjdks/jdk8 /home/jenkins/bootjdks/jdk11 /home/jenkins/bootjdks/jdk17 \
134c134
<  && tar -xzf jdk16.tar.gz --directory=/home/jenkins/bootjdks/jdk16/ --strip-components=1 \
---
>  && tar -xzf jdk17.tar.gz --directory=/home/jenkins/bootjdks/jdk17/ --strip-components=1 \
```

**Review by dsouzai — APPROVED** [review]

**Pull request merged**